data/data.go

```diff
@@ -58,11 +58,35 @@ const (
 
 // StringModifiers denote the status of the possible modifiers for strings
 type StringModifiers struct {
-	Nocase   bool `json:"nocase"`
-	ASCII    bool `json:"ascii"`
-	Wide     bool `json:"wide"`
-	Fullword bool `json:"fullword"`
-	Xor      bool `json:"xor"`
-	I        bool `json:"i"` // for regex
-	S        bool `json:"s"` // for regex
+	Nocase   bool     `json:"nocase"`
+	ASCII    bool     `json:"ascii"`
+	Wide     bool     `json:"wide"`
+	Fullword bool     `json:"fullword"`
+	Private  bool     `json:"private"`
+	Xor      bool     `json:"xor"`
+	XorRange XorRange `json:"xor_range"` // makes sense only with XOR modifier
+	I        bool     `json:"i"`         // for regex
+	S        bool     `json:"s"`         // for regex
+}
+
+// NumBase is used to differentiate between hexadecimal, decimal and octal nums
+type NumBase uint8
+
+// Type of number
+const (
+	Hex NumBase = iota
+	Decimal
+	Octal
+)
+
+// Number specifies range bound for strings XOR modifier
+type Number struct {
```

Comment:
I can't get behind the `Number` type right now, as it's going to break some of my code, as well. It's an issue that I've been wanting to address, and I think that it deserves its own PR. Can you save this code and open a new PR for it?